zstacklib/zstacklib/utils/lv_protocol.py

```
@@ -0,0 +1,437 @@
+        if val % ALIGNMENT != 0:  # must be 4KB-aligned for O_DIRECT
+            return _invalid_header()
+
+    if slot_a_offset < ALIGNMENT:  # must be after header block
+        return _invalid_header()
+    if slot_b_offset <= slot_a_offset:  # B must follow A
+        return _invalid_header()
+    if slot_b_offset < slot_a_offset + slot_a_capacity:  # B must not overlap A
+        return _invalid_header()
+
+    last_update_time = d.get('lastUpdateTime', 0)
+    if not isinstance(last_update_time, _integer_types + (float,)) or last_update_time < 0:
```

Ruff (0.15.9)

[warning] Consider `(*_integer_types, float)` instead of concatenation

Replace with `(*_integer_types, float)`

(RUF005)